policy/modules/services/container.te

```diff
@@ -0,0 +1,880 @@
+# Declarations
+#
+
+attribute container_domain;
+attribute container_net_domain;
+attribute container_user_domain;
+attribute privileged_container_domain;
+
+attribute container_mountpoint_type;
+
+attribute_role container_roles;
+roleattribute system_r container_roles;
+
+container_domain_template(container)
+typealias container_t alias svirt_lxc_net_t;
+typeattribute container_t container_net_domain, container_user_domain;
+
+type spc_t, container_domain, privileged_container_domain;
+domain_type(spc_t)
+role system_r types spc_t;
+
+type spc_user_t, container_domain, container_user_domain, privileged_container_domain;
+domain_type(spc_user_t)
+
+attribute container_engine_domain;
+attribute container_engine_exec_type;
+
+# container engines run by the system
+attribute container_engine_system_domain;
+# container engines run by users
+attribute container_engine_user_domain;
+
+container_engine_domain_template(container_engine)
+typeattribute container_engine_t container_engine_system_domain;
+type container_engine_exec_t, container_engine_exec_type;
+application_domain(container_engine_t, container_engine_exec_t)
+ifdef(`enable_mcs',`
+	init_ranged_daemon_domain(container_engine_t, container_engine_exec_t, s0 - mcs_systemhigh)
+')
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(container_engine_t, container_engine_exec_t, s0 - mls_systemhigh)
+')
+mls_trusted_object(container_engine_t)
+
+container_engine_domain_template(docker)
+typeattribute docker_t container_engine_system_domain;
+type docker_exec_t, container_engine_exec_type;
+application_domain(docker_t, docker_exec_t)
+ifdef(`enable_mcs',`
+	init_ranged_daemon_domain(docker_t, docker_exec_t, s0 - mcs_systemhigh)
+')
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(docker_t, docker_exec_t, s0 - mls_systemhigh)
+')
+mls_trusted_object(docker_t)
+
+container_engine_domain_template(podman)
+typeattribute podman_t container_engine_system_domain;
+type podman_exec_t, container_engine_exec_type;
+application_domain(podman_t, podman_exec_t)
+ifdef(`enable_mcs',`
+	init_ranged_daemon_domain(podman_t, podman_exec_t, s0 - mcs_systemhigh)
+')
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(podman_t, podman_exec_t, s0 - mls_systemhigh)
+')
+mls_trusted_object(podman_t)
+
+container_engine_domain_template(podman_user)
+typeattribute podman_user_t container_engine_user_domain;
+application_domain(podman_user_t, podman_exec_t)
+mls_trusted_object(podman_user_t)
+
+type conmon_exec_t;
+application_executable_file(conmon_exec_t)
+can_exec(podman_t, conmon_exec_t)
+can_exec(podman_user_t, conmon_exec_t)
+allow podman_t conmon_exec_t:file entrypoint;
+allow podman_user_t conmon_exec_t:file entrypoint;
+ifdef(`enable_mcs',`
+	range_transition podman_t conmon_exec_t:process s0;
+	range_transition podman_user_t conmon_exec_t:process s0;
+')
+ifdef(`enable_mls',`
+	range_transition podman_t conmon_exec_t:process mls_systemlow;
+	range_transition podman_user_t conmon_exec_t:process mls_systemlow;
+')
+
```

Comment:
Please move the docker and podman types into two separate modules It also would be better to have conmon_exec_t -> podman_conmon_exec_t